io_scene_niftools/modules/nif_import/armature/__init__.py

```diff
@@ -139,14 +139,16 @@ def store_bind_matrices(self, n_armature):
                                 bonedata.set_transform(diff.get_inverse(fast=False) * bonedata.get_transform())
                         # transforming verts helps with nifs where the skins differ, eg MW vampire or WLP2 Gastornis
                         if isinstance(geom, NifClasses.BSTriShape):
+                            vertex_data = geom.get_vertex_data()
                             if isinstance(geom, NifClasses.BSDynamicTriShape):
                                 # BSDynamicTriShape uses Vector4 to store vertices with a 0 W component, which would
                                 # nullify translation when multiplied by a Matrix44. Hence, first conversion to Vector3
                                 # and assign the position values back later.
                                 vertices = [NifClasses.Vector3.from_value((vertex.x, vertex.y, vertex.z)) for vertex in geom.vertices]
                             else:
-                                vertices = [vert_data.vertex for vert_data in geom.skin.skin_partition.vertex_data]
-                            normals = [vert_data.normal for vert_data in geom.skin.skin_partition.vertex_data]
+                                vertices = [vert_data.vertex for vert_data in vertex_data]
```

Comment:
might be a good idea to vectorize this with numpy?

Reply:
Do you mean the extraction of the data or make the actual vertex_data a numpy array in the codegen?

Reply:
extract the data to an array and then do the matrix math that comes later on that. should be significantly faster than pyffi
having it read as an np array in codegen would be nice too, but that's another issue